dask/dataframe/tests/test_dataframe.py

```diff
@@ -1540,32 +1541,44 @@ def test_dataframe_quantile(method, expected):
     )
     ddf = dd.from_pandas(df, 3)
 
-    with check_numeric_only_deprecation():
-        result = ddf.quantile(method=method)
-    assert result.npartitions == 1
-    assert result.divisions == ("A", "X")
+    numeric_only_kwarg = {}
+    if numeric_only is not None:
+        numeric_only_kwarg = {"numeric_only": numeric_only}
 
-    result = result.compute()
-    assert isinstance(result, pd.Series)
-    assert result.name == 0.5
-    tm.assert_index_equal(result.index, pd.Index(["A", "X", "B"]))
-    assert (result == expected[0]).all()
+    if numeric_only is False or (PANDAS_GT_200 and numeric_only is None):
```

Comment:
I opted for an explicit fail branch to state exactly what errors I'm expecting (`TypeError` in Pandas, but `NotImplementedError` in Dask), but this can be an `xfail` instead, what do you think @jrbourbeau ?